### PR TITLE
Improve zoo layer and rename in tourism attraction

### PR DESCRIPTION
**Pull request by yjacolin**
* split layer into several classes
* add icons for each class
* add cluster feature for "information" features

### CODE REVIEW

**Comment by yjacolin** [comment]
See how it looks like:
gdl_config_default_remote_directory1

See in real life here: http://yjacolin.gis.internal/yjacolin20/wsgi/theme/Paysage

File: mapserver/osm.map.mako [hunk 1]
```diff
@@ -111,6 +107,127 @@ LAYER
...
+    GROUP "zoo" #compatibily
+    EXTENT 473743 74095 839000 306400
+    TYPE POINT
+    STATUS ON
+    TEMPLATE fooOnlyForWMSGetFeatureInfo # For GetFeatureInfo
+    CONNECTIONTYPE postgis
+    PROCESSING "CLOSE_CONNECTION=DEFER" # For performance
+    CONNECTION "user=${dbuser} password=${dbpassword} host=${dbhost} dbname=osm"
+    DATA "geom FROM (SELECT regexp_replace(format(\'%s\', name), \'^$\', osm_id::text) AS display_name,name,osm_id,access,aerialway,amenity,barrier,bicycle,brand,building,covered,denomination,ele,foot,highway,layer,leisure,man_made,motorcar,\"natural\", operator, population, power, place, railway, ref, religion, shop, sport, surface, tourism, waterway, wood, way AS geom FROM planet_osm_point) AS foo USING UNIQUE osm_id USING srid=21781"
+    PROCESSING "NATIVE_FILTER=tourism in ('zoo','information','museum','viewpoint')"
```
Comment (sbrunner) [inline mapserver/osm.map.mako]: Does it creates a where close?

File: mapserver/osm.map.mako [hunk 2]
```diff
@@ -111,6 +107,127 @@ LAYER
...
+    PROJECTION
+        "init=epsg:21781"
+    END
+    TOLERANCE 10
+    TOLERANCEUNITS pixels
+    CLUSTER
+       MAXDISTANCE 40  # in pixels
+       REGION "ellipse"  # can be rectangle or ellipse
+       GROUP ('[tourism]' = 'information') # a logical expression to specify the grouping condition
+     END
```
Comment (sbrunner) [inline mapserver/osm.map.mako]: Indentation issue ...
Shouldn't this block be added also in the layer loop?

**Comment by sbrunner** [comment]
The layer name should also be renamed and the legend role :-)

File: mapserver/symbols.sym [hunk 3]
```diff
@@ -9,6 +9,33 @@ SYMBOLSET
      #ANCHORPOINT 0.5 0.5
   END
 
+  SYMBOL
+    NAME "information"
+    TYPE PIXMAP
+    IMAGE "symbols/information.png"
+    ANCHORPOINT 0.5 1
+  END
+  SYMBOL
```
Comment (sbrunner) [inline mapserver/symbols.sym]: missing end line ...

File: mapserver/osm.map.mako [hunk 4]
```diff
@@ -111,6 +107,127 @@ LAYER
 END
 % endfor
 
+# "column": "tourism",
+# "value": "zoo",
+# "type": "zoo",
+# "name": u"Zoos"
+LAYER
+    NAME "tourism_activity"
+    GROUP "zoo" #compatibily
```
Comment (sbrunner) [inline mapserver/osm.map.mako]: For me we can remove it...

File: mapserver/osm.map.mako [hunk 5]
```diff
@@ -111,6 +107,127 @@ LAYER
...
+    DATA "geom FROM (SELECT regexp_replace(format(\'%s\', name), \'^$\', osm_id::text) AS display_name,name,osm_id,access,aerialway,amenity,barrier,bicycle,brand,building,covered,denomination,ele,foot,highway,layer,leisure,man_made,motorcar,\"natural\", operator, population, power, place, railway, ref, religion, shop, sport, surface, tourism, waterway, wood, way AS geom FROM planet_osm_point) AS foo USING UNIQUE osm_id USING srid=21781"
+    PROCESSING "NATIVE_FILTER=tourism in ('zoo','information','museum','viewpoint')"
+    LABELITEM "name"
+    PROJECTION
+        "init=epsg:21781"
+    END
+    TOLERANCE 10
+    TOLERANCEUNITS pixels
+    CLUSTER
+       MAXDISTANCE 40  # in pixels
```
Comment (sbrunner) [inline mapserver/osm.map.mako]: For me it's too small, I have many information icons and I take many time to see the +xy labels ...

File: mapserver/osm.map.mako (hunk 2 above)
Comment (yjacolin) [inline mapserver/osm.map.mako]: which layer loop?

Comment (sbrunner) [inline mapserver/osm.map.mako]: The END has one more space than the CLUSTER :-)

Comment (sbrunner) [inline mapserver/osm.map.mako]: Here https://github.com/camptocamp/demo_geomapfish/blob/master/mapserver/osm.map.mako#L85 ?

File: mapserver/osm.map.mako (hunk 1 above)
Comment (yjacolin) [inline mapserver/osm.map.mako]: Yes

File: mapserver/osm.map.mako (hunk 2 above)
Comment (yjacolin) [inline mapserver/osm.map.mako]: you want to add the cluster setup to the template LAYER object? This is out of the scope of this PR and I am not sure we want this for all layers from the template. We will discuss this wednesday ;)

File: mapserver/osm.map.mako (hunk 5 above)
Comment (yjacolin) [inline mapserver/osm.map.mako]: yes, the +xy labels is not so clear but I didn't know how to improve this: bigger white buffer around number? Use of another icon (it was my first attempt but it wasn't clear for my workmet)like the one below:

regroup_gray

Any idea?

Comment (sbrunner) [inline mapserver/osm.map.mako]: Looks good :-)

File: mapserver/osm.map.mako (hunk 2 above)
Comment (sbrunner) [inline mapserver/osm.map.mako]: ok :-)

**Pull request merged**